File: restx-samplest/src/main/java/samplest/factory/FactoryInjectionResource.java
package samplest.factory;

import restx.annotations.GET;
import restx.annotations.RestxResource;
import restx.factory.Component;
import restx.security.PermitAll;

import java.util.Set;
import java.util.stream.Collectors;

@Component
@RestxResource("/factory")
public class FactoryInjectionResource {

    Set<TestSetInjection> testSetInjection;

    FactoryInjectionResource(Set<TestSetInjection> testSetInjection) {
        this.testSetInjection = testSetInjection;
    }

    @PermitAll
    @GET("/set")
    public Set<String> testSet() {
        return testSetInjection.stream()
                .map(value -> value.getClass().getName())
                .collect(Collectors.toSet());
    }

}

interface TestSetInjection {
}

@Component
class TestSetInjectionImpl01 implements TestSetInjection {
}

@Component
class TestSetInjectionImpl02 implements TestSetInjection {
}

File: restx-samplest-kotlin/src/main/java/samplest/hello/FactoryInjectionResource.kt
package samplest.hello

import restx.annotations.GET
import restx.annotations.RestxResource
import restx.factory.Component

@Component
@RestxResource("/factory")
class FactoryInjectionResource(
    private val testSetInjection: Set<TestSetInjection>,
    private val testSetAbstract: Set<TestSetAbstract>
) {
    @GET("/interface/set")
    fun testInterfaceSet(): Set<String> = testSetInjection.map { it.javaClass.name }.toSet()

    @GET("/abstract/set")
    fun testAbstractSet(): Set<String> = testSetAbstract.map { it.javaClass.name }.toSet()
}

interface TestSetInjection

@Component
class TestSetInjectionImpl01 : TestSetInjection

@Component
class TestSetInjectionImpl02 : TestSetInjection

abstract class TestSetAbstract

@Component
class TestSetAbstractImpl01 : TestSetAbstract()

@Component
class TestSetAbstractImpl02 : TestSetAbstract()